java-checks/src/main/java/org/sonar/java/checks/SpecializedFunctionalInterfacesCheck.java

```diff
@@ -0,0 +1,239 @@
+          break;
+        case "java.util.function.Supplier":
+        case "java.util.function.Consumer":
+        case "java.util.function.Predicate":
+        case "java.util.function.UnaryOperator":
+        case "java.util.function.BinaryOperator":
+          reportString = handleOneParameterInterface(arguments).map(s -> s + interfaceType.symbolType().name());
+          break;
+        default:
+          return Optional.empty();
+      }
+      if (reportString.isPresent()) {
+        return Optional.of(new InterfaceTreeAndStringPairReport(reportString.get(), interfaceType));
+      }
+    }
+    return Optional.empty();
+  }
+
+  private static Optional<String> handleOneParameterInterface(TypeArguments args) {
+    String reportString = new ParameterTypeNameAndTreeType(args.get(0)).paramTypeName;
+    if (reportString != null) {
+      return Optional.of(reportString);
+    }
+    return Optional.empty();
+  }
+
+  private static Optional<String> handleFunctionInterface(TypeArguments args) {
+    ParameterTypeNameAndTreeType firstArgument = new ParameterTypeNameAndTreeType(args.get(0));
+    ParameterTypeNameAndTreeType secondArgument = new ParameterTypeNameAndTreeType(args.get(1));
+    String reportString = null;
+    if (firstArgument.paramType.equals(secondArgument.paramType)) {
+      reportString = "UnaryOperator<" + firstArgument.paramType.name() + ">";
+    } else if (firstArgument.paramTypeName != null && secondArgument.paramTypeName != null) {
+      reportString = firstArgument.paramTypeName + "To" + secondArgument.paramTypeName + "Function";
+    } else if (firstArgument.paramTypeName == null && secondArgument.paramTypeName != null) {
+      reportString = "To" + secondArgument.paramTypeName + "Function<" + firstArgument.paramType.name() + ">";
+    } else if (firstArgument.paramTypeName != null) {
+      reportString = firstArgument.paramTypeName + "Function<" + secondArgument.paramType.name() + ">";
+    }
+    if (reportString != null) {
+      return Optional.of(reportString);
+    }
+    return Optional.empty();
+  }
+
+  private static Optional<String> handleBiFunctionInterface(TypeArguments args) {
+    ParameterTypeNameAndTreeType firstArgument = new ParameterTypeNameAndTreeType(args.get(0));
+    ParameterTypeNameAndTreeType secondArgument = new ParameterTypeNameAndTreeType(args.get(1));
+    ParameterTypeNameAndTreeType thirdArgument = new ParameterTypeNameAndTreeType(args.get(2));
+    if (firstArgument.paramType.equals(secondArgument.paramType) && firstArgument.paramType.equals(thirdArgument.paramType)) {
+      return Optional.of("BinaryOperator<" + firstArgument.paramType.name() + ">");
+    }
+    return Optional.empty();
+  }
+
+  private static Optional<String> handleBiConsumerInterface(TypeArguments args) {
+    String reportString = returnStringFromJavaObject(((IdentifierTree) args.get(1)).symbolType());
+    if (reportString != null) {
+      return Optional.of("Obj" + reportString + "Consumer");
```

Comment:
And what about the first argument ?!

From what I can see here, `Obj...Consumer` interfaces are paremeterized! https://docs.oracle.com/javase/8/docs/api/java/util/function/ObjIntConsumer.html

It should be something like `ObjIntConsumer<T>`, with T being the other parameter.
Please update the rule as well. (You should have seen that!)